### PR TITLE
Fix telephone validation

### LINKED ISSUE #188: Validation not working if phone number not entered
Phone number validation (or any validation) only works if a value is first entered then removed.

We need to make sure validation works even if no values are submitted.

This resulted in 100s of submissions triggering back-end validation errors. 100s of users would have gotten a sorry page (instead of a validation warning) and we've got many errors triggered via iopipe slack alerts.

### PR DESCRIPTION
**Pull request by AndyEPhipps**
Fixes #188

### CODE REVIEW

File: src/pages/GiftAid/utils/validateForm.js [hunk 1]
```diff
@@ -70,7 +70,7 @@ const validateTransactionId = (donationID) => new RegExp(transactionIdPattern).t
 const getValidation = (validation) => {
   let validity = true;
   Object.keys(validation).map((key) => {
-    if (validation[key].valid !== true && validation[key].valid !== null ) {
+    if (validation[key].valid !== true && validation[key].valid !== "" && validation[key].valid !== null) {
```
Comment (Heleen-cr) [inline src/pages/GiftAid/utils/validateForm.js]: should be `if (validation[key].valid !== true || validation[key].valid === "" || validation[key].valid === null) {
      validity = false;
    }`

Comment (Heleen-cr) [reply src/pages/GiftAid/utils/validateForm.js]: or actually checking for !== true should already cover the other options
I see the code was already checking `if (validation[key].valid !== true && validation[key].valid !== ''){validity = false; } ` which is a bit odd.

File: src/pages/GiftAid/utils/validateForm.js [hunk 2]
```diff
@@ -70,10 +70,11 @@ const validateTransactionId = (donationID) => new RegExp(transactionIdPattern).t
 const getValidation = (validation) => {
   let validity = true;
   Object.keys(validation).map((key) => {
-    if (validation[key].valid !== true && validation[key].valid !== '') {
+    if (validation[key].valid !== true ) {
       validity = false;
```
Comment (Heleen-cr) [inline src/pages/GiftAid/utils/validateForm.js]: can't believe I let that slip through 🤦‍♀

**Comment by AndyEPhipps** [comment]
Will QA this to the best of my ability, in Krupa's absence

**Comment by AndyEPhipps** [comment]
Validation seems happy enough now, will merge this

**Pull request merged**